crates/vm/levm/src/precompiles.rs

```diff
@@ -0,0 +1,182 @@
+    0x00, 0x00, 0x00, 0x07,
+]);
+pub const ECPAIRING_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x08,
+]);
+pub const BLAKE2F_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x09,
+]);
+pub const POINT_EVALUATION_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x0a,
+]);
+
+pub const PRECOMPILES: [H160; 10] = [
+    ECRECOVER_ADDRESS,
+    SHA2_256_ADDRESS,
+    RIPEMD_160_ADDRESS,
+    IDENTITY_ADDRESS,
+    MODEXP_ADDRESS,
+    ECADD_ADDRESS,
+    ECMUL_ADDRESS,
+    ECPAIRING_ADDRESS,
+    BLAKE2F_ADDRESS,
+    POINT_EVALUATION_ADDRESS,
+];
+
+pub fn is_precompile(callee_address: &Address) -> bool {
+    PRECOMPILES.contains(callee_address)
+}
+
+pub fn execute_precompile(current_call_frame: &mut CallFrame) -> Result<(u8, Bytes), VMError> {
+    let callee_address = current_call_frame.code_address;
+    let calldata = current_call_frame.calldata.clone();
+    let gas_for_call = current_call_frame.gas_limit;
+    let consumed_gas = &mut current_call_frame.gas_used;
+
+    let result = match callee_address {
+        address if address == ECRECOVER_ADDRESS => ecrecover(&calldata, gas_for_call, consumed_gas),
+        address if address == IDENTITY_ADDRESS => identity(&calldata, gas_for_call, consumed_gas),
+        address if address == SHA2_256_ADDRESS => sha2_256(&calldata, gas_for_call, consumed_gas),
+        address if address == RIPEMD_160_ADDRESS => {
+            ripemd_160(&calldata, gas_for_call, consumed_gas)
+        }
+        address if address == MODEXP_ADDRESS => modexp(&calldata, gas_for_call, consumed_gas),
+        address if address == ECADD_ADDRESS => ecadd(&calldata, gas_for_call, consumed_gas),
+        address if address == ECMUL_ADDRESS => ecmul(&calldata, gas_for_call, consumed_gas),
+        address if address == ECPAIRING_ADDRESS => ecpairing(&calldata, gas_for_call, consumed_gas),
+        address if address == BLAKE2F_ADDRESS => blake2f(&calldata, gas_for_call, consumed_gas),
+        address if address == POINT_EVALUATION_ADDRESS => {
+            point_evaluation(&calldata, gas_for_call, consumed_gas)
+        }
+        _ => return Err(VMError::Internal(InternalError::InvalidPrecompileAddress)),
+    };
+    match result {
+        Ok(res) => Ok((SUCCESS_FOR_RETURN, res)),
+        Err(_) => {
+            // Maybe we should return an Err in this case. Like differencing between OOG,
+            // errors produced by wrong inputs an internal errors
+            *consumed_gas = consumed_gas
+                .checked_add(gas_for_call)
+                .ok_or(InternalError::ArithmeticOperationOverflow)?;
+            Ok((REVERT_FOR_RETURN, Bytes::new()))
+        }
+    }
```

Comment:
With the change I suggested we can delete this, because every precompile will now propagate their respective errors. The errors should be `VMError`. Inside the `VMError` in this case there will be either an `InternalError` or a `PrecompileError`.

Reply:
Also done in [Change execute_precompile return values](https://github.com/lambdaclass/ethrex/pull/1454/commits/6cb7ade0bade8f3d5e478ad4ca6cc4c41296625a)